### PR TITLE
Fix PostgreSQL backup restore

### LINKED ISSUE #1623: Unable to restore the database in docker version
**Note: for support questions, please use the `cookiecutter-django` tag on stackoverflow**. This repository's issues are reserved for feature requests and bug reports. If you need quick professional paid support for your project, contact support@cookiecutter.io.

* **I'm submitting a ... **
  - [* ] bug report
  - [ ] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

* **Do you want to request a *feature* or report a *bug*?**
report bug

* **What is the current behavior?**
```
INFO: Dropping all connections to the database...
psql: could not connect to server: Connection refused
	Is the server running on host "localhost" (127.0.0.1) and accepting
	TCP/IP connections on port 5432?
could not connect to server: Cannot assign requested address
	Is the server running on host "localhost" (::1) and accepting
	TCP/IP connections on port 5432?


```

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem** 
when I ran the following command
 sudo docker-compose -f local.yml run --rm postgres restore  her.sql.gz
instead of restoring the db I got the above error

* **What is the expected behavior?**
restore should have taken place normally

* **What is the motivation / use case for changing the behavior?**

* **Please tell us about your environment:**
Docker version of the project

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

### PR DESCRIPTION
**Pull request by webyneter**
Fixes #1623.

### CODE REVIEW

File: {{cookiecutter.project_slug}}/compose/production/postgres/maintenance/restore [hunk 1]
```diff
@@ -37,40 +37,18 @@ if [[ "${POSTGRES_USER}" == "postgres" ]]; then
...
-    --dbname=postgres \
-    --command="${drop_postgres_connections_sql}"
+export PGDATABASE="${POSTGRES_DB}"
 
 message_info "Dropping the database..."
-dropdb \
-    --host=postgres \
-    --username="${POSTGRES_USER}" \
-    "${POSTGRES_DB}"
+dropdb "${PGDATABASE}"
```
Comment (browniebroke) [inline {{cookiecutter.project_slug}}/compose/production/postgres/maintenance/restore]: This seems to fail when I try to follow the instructions, there are opened connections. I think I've followed the steps from the documentation:

```bash
> docker-compose -f local.yml up -d
Starting testrestore_postgres_1 ... done
Starting testrestore_django_1   ... done

> docker-compose -f local.yml run --rm postgres backup
Backing up the 'test_restore' database...
SUCCESS: 'test_restore' database backup 'backup_2018_05_01T13_25_26.sql.gz' has been created and placed in '/backups'.

> docker-compose -f local.yml run --rm postgres backups
These are the backups you have got:
total 12K
-rw-r--r-- 1 root root 5.2K May  1 13:25 backup_2018_05_01T13_25_26.sql.gz
-rw-r--r-- 1 root root   20 May  1 13:24 backup_2018_05_01T13_24_46.sql.gz

> docker-compose -f local.yml run --rm postgres restore backup_2018_05_01T13_25_26.sql.gz
Restoring the 'test_restore' database from the '/backups/backup_2018_05_01T13_25_26.sql.gz' backup...
INFO: Dropping the database...
dropdb: database removal failed: ERROR:  database "test_restore" is being accessed by other users
DETAIL:  There are 2 other sessions using the database.
```

Also the [example of output[(https://cookiecutter-django.readthedocs.io/en/latest/docker-postgres-backups.html#restoring-from-the-existing-backup) for the restore is out of date with this change.

Comment (browniebroke) [reply {{cookiecutter.project_slug}}/compose/production/postgres/maintenance/restore]: The last command worked after I did:

```bash
docker-compose -f local.yml stop django
```

File: docs/docker-postgres-backups.rst [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 
-For brevity it is assumed that will be running the below commands against local environment, however, this is by no means mandatory so feel free switching to ``production.yml`` when needed.
+Prerequisites
+-------------
 
-Note that the application stack should not necessarily be running when applying any of the instructions below, unless explicitly stated otherwise. For instance, suppose the stack has been down for quite some time or have never even been up yet -- rather than starting it beforehand use a single ``$ docker-compose -f local.yml run --rm <command>`` with the desired command. By contrast, should you already have your application up and running do not bother waiting for ``run`` instruction to finish (they usually take a bit longer due to bootstrapping phase), just use ``$ docker-compose -f local.yml exec <command>`` instead; note that any ``exec`` command fails unless all of the required containers are running. From now on, we will be using ``run``-style examples for general-case compatibility.
+#. the project was generated with ``use_docker`` set to ``y``;
+#. the stack is up and running: ``docker-compose -f local.yml up -d``.
```
Comment (browniebroke) [inline docs/docker-postgres-backups.rst]: Maybe we should tell to only start the `postgres` service here?

```bash
> docker-compose -f local.yml up -d postgres
Starting testrestore_postgres_1 ... done
```

File: docs/docker-postgres-backups.rst [hunk 3]
```diff
@@ -1,13 +1,14 @@
 PostgreSQL Backups with Docker
 ==============================
 
-Prerequisites:
+.. note:: For brevity it is assumed that you will be running the below commands against local environment, however, this is by no means mandatory so feel free switching to ``production.yml`` when needed.
```
Comment (browniebroke) [inline docs/docker-postgres-backups.rst]: I'm no native English speaker, but isn't it more English to say "feel free to switch"?

Comment (webyneter) [reply docs/docker-postgres-backups.rst]: Me neither, but yours sounds right to me (and [not only me](https://dictionary.cambridge.org/us/dictionary/english/feel-free-to-do-something).)

**Pull request merged**